### PR TITLE
Create e2e tests for Prometheus metrics

### LINKED ISSUE #4127: Create e2e tests for all exposed Prometheus metrics
### Proposal

At the moment KEDA exposes following metrics:

- [x] `keda_scaler_active`
- [x] `keda_scaler_metrics_value`
- [x] `keda_scaler_metrics_latency`
- [ ] `keda_scaler_errors`
- [ ] `keda_scaler_error_totals`
- [ ] `keda_scaled_object_errors`
- [x] `keda_resource_totals`
- [x] `keda_trigger_totals`

We should cover all of these metrics by e2e tests:
 - test that metrics are exposed
 - test that metrics are correctly calculated (ie. we should test that metrics related to errros are correctly incremented in case of an error in a scaler/scaledobject, etc.).

Exisiting Prometheus Metrics e2e tests: https://github.com/kedacore/keda/blob/main/tests/internals/prometheus_metrics/prometheus_metrics_test.go

Docs: https://keda.sh/docs/2.10/operate/prometheus/

### PR DESCRIPTION
**Pull request by Parthiba-Hazra**
write test cases for Prometheus Metrics e2e tests, covered this metrics
- keda_scaler_errors
- keda_scaler_error_totals
- keda_scaled_object_errors

test cases check that Prometheus metrics are correctly incremented in case of an error in a scaler/scaledobject, etc.

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4127 

Relates to #4127

### CODE REVIEW

File: tests/internals/prometheus_metrics/prometheus_metrics_test.go [hunk 1]
```diff
@@ -230,6 +256,9 @@ func TestPrometheusMetrics(t *testing.T) {
 	testScalerMetricValue(t)
 	testScalerMetricLatency(t)
 	testScalerActiveMetric(t)
+	testSclaedObjectErrors(t, data)
+	testSclaerErrors(t)
```
Comment (JorTurFer) [inline tests/internals/prometheus_metrics/prometheus_metrics_test.go:260]:
```suggestion
	testScalerErrors(t)
```

File: tests/internals/prometheus_metrics/prometheus_metrics_test.go [hunk 2]
```diff
@@ -292,6 +322,109 @@ func testScalerMetricValue(t *testing.T) {
...
+			assert.Greater(t, errCounterVal2, errCounterVal1)
+		} else {
+			t.Errorf("metric not available")
+		}
+	} else {
+		t.Errorf("metric not available")
+	}
+}
+
+func testSclaerErrors(t *testing.T) {
```
Comment (JorTurFer) [inline tests/internals/prometheus_metrics/prometheus_metrics_test.go:350]:
```suggestion
func testScalerErrors(t *testing.T) {
```

File: tests/internals/prometheus_metrics/prometheus_metrics_test.go [hunk 3]
```diff
@@ -292,6 +322,109 @@ func testScalerMetricValue(t *testing.T) {
...
+		} else {
+			t.Errorf("metric not available")
+		}
+	} else {
+		t.Errorf("metric not available")
+	}
+}
+
+func testScalerErrorsTotal(t *testing.T, data templateData) {
+	t.Log("--- testing sclaer errors toatl ---")
```
Comment (JorTurFer) [inline tests/internals/prometheus_metrics/prometheus_metrics_test.go:373]:
```suggestion
	t.Log("--- testing scaler errors total ---")
```

File: tests/internals/prometheus_metrics/prometheus_metrics_test.go [hunk 4]
```diff
@@ -292,6 +322,109 @@ func testScalerMetricValue(t *testing.T) {
...
+		} else {
+			t.Errorf("metric not available")
+		}
+	} else {
+		t.Errorf("metric not available")
+	}
+}
+
+func testSclaerErrors(t *testing.T) {
+	t.Log("--- testing sclaer errors ---")
```
Comment (JorTurFer) [inline tests/internals/prometheus_metrics/prometheus_metrics_test.go:351]:
```suggestion
	t.Log("--- testing scaler errors ---")
```

**Review by JorTurFer — COMMENTED** [review]
Thanks for the contribution!

I have seen that you are only deploying the wrong scaledobject in the first test case and other cases are assuming that there is an ScaledObject with an error from a previous test case. 
Test cases should be independent and they can't assume any status from other test. 
Could you add the required lines to deploy the wrong scaledobject on each test case?  This won't change anything from test pov but will guarantee that those tests work if we remove the previous one or we reorder them

**Comment by Parthiba-Hazra** [comment]
Thanks @JorTurFer for your review. I will make those changes

**Comment by JorTurFer** [comment]
Hi @Parthiba-Hazra ,
Could you rebase your PR? `prometheus_metrics_test.go` file has been moved from internals to sequential folder

**Comment by JorTurFer** [comment]
Hi!
No worries xD DCO is the check that we use to ensure that all the commits are signed. I guess that any commit isn't signed and that's why it's failing. In [this link](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what) you can see how to solve it.
Let me know if you have any doubt about it 😄

**Comment by JorTurFer** [comment]
Hi @Parthiba-Hazra 
DCO is still failing, could you sign your commits please? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md?rgh-link-date=2023-04-03T14%3A50%3A17Z#i-didnt-sign-my-commit-now-what

**Comment by Parthiba-Hazra** [comment]
> Hi @Parthiba-Hazra DCO is still failing, could you sign your commits please? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md?rgh-link-date=2023-04-03T14%3A50%3A17Z#i-didnt-sign-my-commit-now-what

I follow the steps but it wont work I think.

**Comment by Parthiba-Hazra** [comment]
Hi @JorTurFer 
I also notice that keda_sclaer_errors test failed, I run same test case in my local minikube cluster but it passed the check.

**Comment by Parthiba-Hazra** [comment]
> Hi @Parthiba-Hazra DCO is still failing, could you sign your commits please? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md?rgh-link-date=2023-04-03T14%3A50%3A17Z#i-didnt-sign-my-commit-now-what

Hey I messed up it, actually I shouldn't place a PR from my main branch. Should I close this PR and create a new PR with same changes?

**Comment by zroubalik** [comment]
> > Hi @Parthiba-Hazra DCO is still failing, could you sign your commits please? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md?rgh-link-date=2023-04-03T14%3A50%3A17Z#i-didnt-sign-my-commit-now-what
> 
> Hey I messed up it, actually I shouldn't place a PR from my main branch. Should I close this PR and create a new PR with same changes?

Sure, not a problem. I will close this PR.

**Pull request closed without merging**